packages/pigeon/lib/dart_generator.dart

```diff
@@ -137,9 +153,12 @@ final BinaryMessenger$nullTag _binaryMessenger;
             '\'$channelName\', codec, binaryMessenger: _binaryMessenger);',
           );
         });
-        final String returnStatement = func.returnType == 'void'
+        final String returnType =
+            _makeGenericTypeArguments(func.returnType, nullTag);
+        final String castCall = _makeGenericCastCall(func.returnType, nullTag);
+        final String returnStatement = func.returnType.dataType == 'void'
```

Comment:
Yea, we've talked about this in the past.  I plan on refactoring this.  This wasn't introduced in this PR, just continues with the convention that was already set.

Reply:
I know it wasn't introduced in this PR, but `dataType` being a class that you can trivially add an `isVoid` method to is new in this PR. Why not include that one extra line in the new class in order to improve all of these lines that you have to change in this PR anyway?

Reply:
The diff is easier to verify since the logic isn't changing.  I'll do this in a follow up right now.  Shouldn't take but a second and this PR is already big.

Reply:
https://github.com/flutter/packages/pull/432